File: Day4/main.ts
function createXmasTree(height: number, ornament: string): string {
  const width = height * 2 - 1;
  const baseLine = '_'.repeat((width - 1) / 2) + '#' + '_'.repeat((width - 1) / 2);
  const tree = Array.from({ length: height }, (_, i) => {
    const ornamentCount = i * 2 + 1;
    const padding = (width - ornamentCount) / 2;
    return '_'.repeat(padding) + ornament.repeat(ornamentCount) + '_'.repeat(padding);
  });
  return [...tree, baseLine, baseLine].join('\n');
}

File: Day4/main.py
def create_xmas_tree(height: int, ornament: str) -> str:
    width = height * 2 - 1
    base_line = ('_' * ((width - 1) // 2)) + '#' + ('_' * ((width - 1) // 2))

    tree = []
    for i in range(height):
        num_ornaments = i * 2 + 1
        spaces = (width - num_ornaments) // 2
        tree.append('_' * spaces + ornament * num_ornaments + '_' * spaces)

    return '\n'.join(tree + [base_line, base_line])

